modules/secret_manager/main.tf

```diff
@@ -0,0 +1,49 @@
+resource "random_password" "random_pass" {
+  length  = 16
+  special = false
+}
+
+# Recover the GCP secret payload when GCP secret name is provided
+data "google_secret_manager_secret_version" "gcp_predefined_pass" {
+  secret    = var.secret_value
+  project   = var.project
+}
+
+locals {
+   secret_value = var.secret_value == "" ? random_password.random_pass.result : data.google_secret_manager_secret_version.gcp_predefined_pass.secret_data
+}
+
+# GCP Secret Manager
+resource "google_secret_manager_secret" "secret" {
+  project     = var.project
+  secret_id   = var.secret_id
+  labels      = var.secret_labels
+  expire_time = var.secret_expire_time
+
+  replication {
+    user_managed {
+      replicas {
+        location = var.region
+      }
+    }
+  }
+  count = var.secret_value == "" ? 1 : 0
+}
+
+# GCP Secret Manager Payload
+resource "google_secret_manager_secret_version" "secret" {
```

Comment:
The _parent_ resource `google_secret_manager_secret.secret[0]` will be created only if `var.secret_value == ""`, so I think that we need the same condition here.